components/brave_vpn/common/mojom/brave_vpn.mojom

```diff
@@ -73,6 +73,14 @@ interface ServiceHandler {
   [EnableIfNot=is_android]
   ResetConnectionState();
 
+  [EnableIfNot=is_android]
+  EnableOnDemand(bool enable);
+
+  [EnableIfNot=is_android]
+  // true to |available| when on demand feature is supported.
+  // |enabled| is only valid when |available| is true.
+  GetOnDemandState() => (bool available, bool enabled);
```

Comment:
I'm curious why the `available` parameter is needed, because `BraveVpnService::GetOnDemandState()` always passes `true` for it.

Reply:
`available` is set to `false` for on macOS platform as this feature is used only on macOS now.
Two additional methods are added for non-platforms.
With `available`, UI doesn't need to have platform guard.
I thought having platform specific code in c++ logic instead of webui.

Reply:
Makes sense. Thanks.